File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-0.py
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if not root:
            return TreeNode(val)
        cur = root

        while True:
            if val > cur.val and cur.right is not None:
                cur = cur.right
            elif val > cur.val and cur.right is None:
                cur.right = TreeNode(val)
                break
            elif val < cur.val and cur.left is not None:
                cur = cur.left
            elif val < cur.val and cur.left is None:
                cur.left = TreeNode(val)
                break
        return root

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cs
public class Solution
{
	public List<List<int>> LevelOrder(TreeNode root)
	{
		List<List<int>> output = new List<List<int>>();
		
		if(root == null)
			return output;
		
		Queue<TreeNode> queue = new Queue<TreeNode>();
		queue.Enqueue(root);
			
		while(queue.Count > 0)
		{
			int level = queue.Count();
			List<int> levelList = new List<int>();
			
			for (int i = 0; i < level; i++)
			{
				var current = queue.Dequeue();
				levelList.Add(current.val);

				// First queue the left
				if (current.left != null)
					queue.Enqueue(current.left);

				// Then queue the right - so the stuff at the left comes before the stuff at the right
				if(current.right != null)
					queue.Enqueue(current.right);
			}
			output.Add(levelList);
		}	
		
		return output;
	}
}
